cmd/livepeer/livepeer.go

```diff
@@ -330,6 +341,49 @@ func main() {
 		// Setup unbonding service to manage unbonding locks
 		n.EthServices["UnbondingService"] = eventservices.NewUnbondingService(n.Eth, dbh)
 
+		// Initialize block watcher that will emit logs used by event watchers
+		blockWatcherClient, err := blockwatch.NewRPCClient(*ethUrl, ethRPCTimeout)
+		if err != nil {
+			glog.Errorf("Failed to setup blockwatch client: %v", err)
+			return
+		}
+		topics := watchers.FilterTopics()
+		blockWatcherCfg := blockwatch.Config{
+			Store:               n.Database,
+			PollingInterval:     blockWatcherPollingInterval,
+			StartBlockDepth:     rpc.LatestBlockNumber,
+			BlockRetentionLimit: blockWatcherRetentionLimit,
+			WithLogs:            true,
+			Topics:              topics,
+			Client:              blockWatcherClient,
+		}
+		// Wait until all event watchers have been initialized before starting the block watcher
+		blockWatcher := blockwatch.New(blockWatcherCfg)
+
+		blockWatchCtx, cancel := context.WithCancel(context.Background())
```

Comment:
Let's move the logic for backfilling and starting the block watcher to the end of the if clause for on-chain mode so that we can ensure that we always start the event watchers before the block watcher even if certain event watchers in the future are specific to a particular node type (i.e. orchestrator or broadcaster only). See the [UnbondingWatcher PR](https://github.com/livepeer/go-livepeer/pull/1048/files#diff-205b550ef0ac99b368da16cfb883960bR497).

Reply:
fixed in 5399d9d